pallets/did/src/lib.rs

```diff
@@ -172,6 +172,7 @@ pub mod pallet {
 		type Origin: From<DidRawOrigin<DidIdentifierOf<Self>>>;
 		#[cfg(feature = "runtime-benchmarks")]
 		type Origin: From<RawOrigin<DidIdentifierOf<Self>>>;
+		type EnsureOrigin: EnsureOrigin<Success = DidIdentifierOf<Self>, <Self as frame_system::Config>::Origin>;
```

Comment:
Do we need a generic origin? Can we not simply use the did origin that we defined in this crate?